src/v/config/configuration.cc

```diff
@@ -474,6 +474,26 @@ configuration::configuration()
        .visibility = visibility::user},
       model::timestamp_type::create_time,
       {model::timestamp_type::create_time, model::timestamp_type::append_time})
+  , log_message_timestamp_alert_before_ms(
+      *this,
+      "log_message_timestamp_alert_before_ms",
+      "Threshold in milliseconds for alerting on messages with a timestamp "
+      "before the broker "
+      "time, as in the past from the point of view of the broker",
+      {.needs_restart = needs_restart::no,
+       .example = "315360000000",
+       .visibility = visibility::tunable},
+      std::chrono::milliseconds::max())
+  , log_message_timestamp_alert_after_ms(
+      *this,
+      "log_message_timestamp_alert_after_ms",
+      "Threshold in milliseconds for alerting on messages with a timestamp "
+      "after the broker "
+      "time, as in the future from the point of view of the broker",
+      {.needs_restart = needs_restart::no,
+       .example = "3600000",
+       .visibility = visibility::tunable},
```

Comment:
Should this be `visibility::user`? Reading the descriptions of each i'm not sure, but `log_message_timestamp_type` is user-visible so perhaps we just follow that. OTOH, it affects system behavior like GCing, so maybe tunable is the way to go

Reply:
well the configs are only to increment the metric and to log a warning. after the overall word is completed, gc will not use message timestamps anymore, and this pr in isolation does not modify the current behaviour.
i'd keep it as tunable mostly to reduce the cognitive overload for the basic usage, i would not expect for people to modify the values

Reply:
> i'd keep it as tunable mostly to reduce the cognitive overload for the basic usage, i would not expect for people to modify the values

+1 also the metrics here are internal metrics not the public ones